src/readers/morphologyASC.cpp

```diff
@@ -76,23 +76,30 @@ class NeurolucidaParser
     }
 
   private:
-    std::tuple<Point, floatType> parse_point(NeurolucidaLexer& lex) {
+    std::tuple<Point, floatType> parse_point(NeurolucidaLexer& lex, bool is_marker) {
         lex.expect(Token::LPAREN, "Point should start in LPAREN");
-        std::array<morphio::floatType, 4> point{};  // X,Y,Z,R
-        for (auto& p : point) {
+        std::array<morphio::floatType, 4> point{};  // X,Y,Z,D
+        for (unsigned int i = 0; i < 4; i++) {
             try {
 #ifdef MORPHIO_USE_DOUBLE
-                p = std::stod(lex.consume()->str());
+                point[i] = std::stod(lex.consume()->str());
 #else
-                p = std::stof(lex.consume()->str());
+                point[i] = std::stof(lex.consume()->str());
 #endif
             } catch (const std::invalid_argument&) {
                 throw RawDataError(err_.ERROR_PARSING_POINT(lex.line_num(), lex.current()->str()));
             }
+
+            // Markers can have an s-exp (X Y Z) without diameter
+            if (is_marker && i == 2 && (lex_.peek()->str() == ")")) {
```

Comment:
haha ok. There is no other way than `")"` to check you reached the end ? Or something like a lex_.size() == 3 ?

Reply:
It's a parser so you don't know what's ahead of you, you can simply "peek" at the next token :)
Otherwise that would be too easy.

Reply:
pouet